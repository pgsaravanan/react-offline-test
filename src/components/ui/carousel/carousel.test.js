import React from 'react';
import { mount } from 'enzyme';
import Carousel from './carousel';
import 'jest-styled-components';

const data = {
  hits: [
    { id: 1, webformatURL: 'xyz' },
    { id: 2, webformatURL: 'xyz1' },
    { id: 3, webformatURL: 'xyz2' }
  ]
};

describe('Carousel', () => {
  it('should match snapshot', () => {
    const tree = mount(<Carousel data={data} />);
    expect(tree).toMatchSnapshot();
  });

  it('should contain datain carousel in list format', () => {
    const tree = mount(<Carousel data={data} />);

    expect(tree.find('.carousel-container').find('li').length).toEqual(3);
  });

  it('should set the container width based on number of items in carousel', () => {
    const tree = mount(<Carousel data={data} />);

    expect(tree.find('.carousel-container').find('ul')).toHaveStyleRule('width', '1350px');
  });

  it('should set the container style margin-left  as -0px by default', () => {
    const tree = mount(<Carousel data={data} />);

    expect(tree.find('.carousel-container').find('ul')).toHaveStyleRule('margin-left', '-0px');
  });

  describe('On Next click', () => {
    it('should update margin-left property to move next slide', () => {
      const tree = mount(<Carousel data={data} />);

      tree.find('.carousel-button-wrapper').find('.next').simulate('click');

      expect(tree.find('.carousel-container').find('ul')).toHaveStyleRule('margin-left', '-470px');
    });

    it('should update margin-left property to move next slide and click next button 2 times', () => {
      const tree = mount(<Carousel data={data} />);

      tree.find('.carousel-button-wrapper').find('.next').simulate('click');

      tree.find('.carousel-button-wrapper').find('.next').simulate('click');

      expect(tree.find('.carousel-container').find('ul')).toHaveStyleRule('margin-left', '-980px');
    });
  });

  describe('On Previous click', () => {
    it('should update margin-left property to move previous slide', () => {
      const tree = mount(<Carousel data={data} />);
      tree.find('.carousel-button-wrapper').find('.next').simulate('click');
      tree.find('.carousel-button-wrapper').find('.next').simulate('click');
      tree.find('.carousel-button-wrapper').find('.next').simulate('click');

      tree.find('.carousel-button-wrapper').find('.prev').simulate('click');

      expect(tree.find('.carousel-container').find('ul')).toHaveStyleRule('margin-left', '-980px');

      tree.find('.carousel-button-wrapper').find('.prev').simulate('click');

      expect(tree.find('.carousel-container').find('ul')).toHaveStyleRule('margin-left', '-470px');

      tree.find('.carousel-button-wrapper').find('.prev').simulate('click');

      expect(tree.find('.carousel-container').find('ul')).toHaveStyleRule('margin-left', '-0px');
    });
  });
});
