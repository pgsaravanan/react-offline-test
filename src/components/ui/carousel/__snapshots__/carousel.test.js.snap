// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should match snapshot 1`] = `
.c0 {
  margin: 20px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  width: 1350px;
  margin-left: -0px;
}

.c0 li {
  padding: 0 20px;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c0 li img {
  height: 300px;
  width: 450px;
}

.c0 li:nth-child(3) img {
  border: solid 5px black;
}

@media (min-width:768px) {
  .c0 {
    width: 450px;
    margin-left: -0px;
  }
}

@media screen and (min-width:768px) {
  .c0 li img {
    width: 150px;
    height: 100px;
  }
}

<Carousel
  data={
    Object {
      "hits": Array [
        Object {
          "id": 1,
          "webformatURL": "xyz",
        },
        Object {
          "id": 2,
          "webformatURL": "xyz1",
        },
        Object {
          "id": 3,
          "webformatURL": "xyz2",
        },
      ],
    }
  }
>
  <div>
    <div
      className="carousel-container"
    >
      <styled.ul
        active={2}
        index={0}
      >
        <ul
          className="c0"
        >
          <li
            key="1"
          >
            <img
              alt="xyz"
              src="xyz"
            />
          </li>
          <li
            key="2"
          >
            <img
              alt="xyz1"
              src="xyz1"
            />
          </li>
          <li
            key="3"
          >
            <img
              alt="xyz2"
              src="xyz2"
            />
          </li>
        </ul>
      </styled.ul>
      <div
        className="carousel-button-wrapper"
      >
        <button
          aria-label="Previous"
          className="prev"
          onClick={[Function]}
        />
        <button
          aria-label="Next"
          className="next"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</Carousel>
`;
